init/eessi_archdetect.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# current pathways implemented in EESSI 2021.12 pilot version
+# x86_64/generic
+# x86_64/intel/haswell
+# x86_64/intel/skylake_avx512
+# x86_64/amd/zen2
+# x86_64/amd/zen3
+# aarch64/generic
+# aarch64/graviton2
+# aarch64/graviton3
+# ppc64le/generic
+# ppc64le/power9le
+
+ARGUMENT=${1:-none}
+
+cpupath () {
+  # let the kernel tell base machine type
+  MACHINE_TYPE=${EESSI_MACHINE_TYPE:-$(uname -m)}
+  PROC_CPUINFO=${EESSI_PROC_CPUINFO:-/proc/cpuinfo}
+  
+  # clean up any existing pointers to cpu features
+  unset $(env | grep EESSI_HAS | cut -f1 -d=)
+  unset EESSI_CPU_VENDOR
+
+  # fallback path
+  CPU_PATH="${MACHINE_TYPE}/generic"
+
+  if [ ${MACHINE_TYPE} == "aarch64" ]; then
+    CPUINFO_VENDOR_FLAG=$(grep -m 1 -i ^vendor ${PROC_CPUINFO})
+    [[ $CPUINFO_VENDOR_FLAG =~ "ARM" ]] && EESSI_CPU_VENDOR=arm
+
+    if [ ${EESSI_CPU_VENDOR} == "arm" ]; then
+      CPU_FLAGS=$(grep -m 1 -i ^flags ${PROC_CPUINFO} | sed 's/$/ /g')
+      [[ $CPU_FLAGS =~ " asimd " ]] && EESSI_HAS_ASIMD=true
+      [[ $CPU_FLAGS =~ " svei8mm " ]] && EESSI_HAS_SVEI8MM=true
+
+      [[ ${EESSI_HAS_ASIMD} ]] && EESSI_CPU_TYPE=neoverse-n1
+      [[ ${EESSI_HAS_SVEI8MM} ]] && EESSI_CPU_TYPE=neoverse-v1
+      [[ $EESSI_CPU_TYPE ]] && CPU_PATH="${MACHINE_TYPE}/${EESSI_CPU_VENDOR}/${EESSI_CPU_TYPE}"
+    else
+      CPU_FLAGS=$(grep -m 1 -i ^features ${PROC_CPUINFO} | sed 's/$/ /g')
+      [[ $CPU_FLAGS =~ " asimd " ]] && EESSI_HAS_ASIMD=true
+      [[ $CPU_FLAGS =~ " svei8mm " ]] && EESSI_HAS_SVEI8MM=true
+
+      [[ ${EESSI_HAS_ASIMD} ]] && EESSI_CPU_TYPE=graviton2
+      [[ ${EESSI_HAS_SVEI8MM} ]] && EESSI_CPU_TYPE=graviton3
+      [[ $EESSI_CPU_TYPE ]] && CPU_PATH="${MACHINE_TYPE}/${EESSI_CPU_TYPE}"
+    fi
```

Comment:
I would never return `graviton2` as a result here, since then we get get `graviton2` as a result while we're actually on a `thunderx2`, for example. We should break here with `archspec`, and symlink `aarch64/arm/neoverse-n1` to `aarch64/graviton2` in EESSI pilot 2021.12, and then the reverse in the next EESSI pilot (so it still works with `archspec`).

```shell
  if [ ${MACHINE_TYPE} == "aarch64" ]; then
    CPUINFO_VENDOR_FLAG=$(grep -m 1 -i ^vendor ${PROC_CPUINFO})
    [[ $CPUINFO_VENDOR_FLAG =~ "ARM" ]] && EESSI_CPU_VENDOR=arm

   if [ ${EESSI_CPU_VENDOR} == "arm" ]; then
      CPU_FLAGS=$(grep -m 1 -i ^flags ${PROC_CPUINFO} | sed 's/$/ /g')
   else
      CPU_FLAGS=$(grep -m 1 -i ^features ${PROC_CPUINFO} | sed 's/$/ /g')
   fi
   [[ $CPU_FLAGS =~ " asimd " ]] && EESSI_HAS_ASIMD=true
   [[ $CPU_FLAGS =~ " svei8mm " ]] && EESSI_HAS_SVEI8MM=true

   [[ ${EESSI_HAS_ASIMD} ]] && EESSI_CPU_TYPE=neoverse-n1
   [[ ${EESSI_HAS_SVEI8MM} ]] && EESSI_CPU_TYPE=neoverse-v1
   [[ $EESSI_CPU_TYPE ]] && CPU_PATH="${MACHINE_TYPE}/${EESSI_CPU_VENDOR}/${EESSI_CPU_TYPE}"
```

Reply:
implemented